src/parser/column_definition.cpp

```diff
@@ -1,39 +1,174 @@
 #include "duckdb/parser/column_definition.hpp"
 #include "duckdb/common/field_writer.hpp"
+#include "duckdb/parser/parsed_expression_iterator.hpp"
+#include "duckdb/parser/expression/columnref_expression.hpp"
+#include "duckdb/parser/parsed_data/alter_table_info.hpp"
+#include "duckdb/parser/expression/cast_expression.hpp"
 
 namespace duckdb {
 
 ColumnDefinition::ColumnDefinition(string name_p, LogicalType type_p) : name(move(name_p)), type(move(type_p)) {
 }
 
-ColumnDefinition::ColumnDefinition(string name_p, LogicalType type_p, unique_ptr<ParsedExpression> default_value)
-    : name(move(name_p)), type(move(type_p)), default_value(move(default_value)) {
+ColumnDefinition::ColumnDefinition(string name_p, LogicalType type_p, ColumnExpression expression)
+    : name(move(name_p)), type(move(type_p)) {
+	switch (expression.type) {
+	case ColumnExpressionType::DEFAULT: {
+		default_value = move(expression.expression);
+		break;
+	}
+	case ColumnExpressionType::GENERATED: {
+		generated_expression = move(expression.expression);
+		category = TableColumnType::GENERATED;
+		break;
+	}
+	default: {
+		throw InternalException("Type not implemented for ColumnExpressionType");
+	}
+	}
 }
 
 ColumnDefinition ColumnDefinition::Copy() const {
 	ColumnDefinition copy(name, type);
 	copy.oid = oid;
+	copy.storage_oid = storage_oid;
 	copy.default_value = default_value ? default_value->Copy() : nullptr;
+	copy.generated_expression = generated_expression ? generated_expression->Copy() : nullptr;
 	copy.compression_type = compression_type;
+	copy.category = category;
 	return copy;
 }
 
 void ColumnDefinition::Serialize(Serializer &serializer) const {
 	FieldWriter writer(serializer);
 	writer.WriteString(name);
 	writer.WriteSerializable(type);
-	writer.WriteOptional(default_value);
+	writer.WriteField<TableColumnType>(category);
+	if (Generated()) {
+		writer.WriteOptional(generated_expression);
```

Comment:
This will break the storage version, move `WriteField` to the end, and use `ReadField` below see #2969